js/common/tests/shell/shell-community-sharding-cluster.js

```diff
@@ -139,126 +156,147 @@ function DocumentShardingSuite() {
       assertEqual(5, c2.properties()["numberOfShards"]);
       assertEqual(["_key"], c1.properties()["shardKeys"]);
       assertEqual(["_key"], c2.properties()["shardKeys"]);
-      
+
       for (let i = 0; i < 1000; ++i) {
         c2.insert({ _key: "test" + i, value: i });
       }
-      
+
       assertEqual([ 188, 192, 198, 204, 218 ], Object.values(c2.count(true)).sort());
 
       for (let i = 0; i < 1000; ++i) {
         assertEqual(i, c2.document("test" + i).value);
       }
     },
-    
+
+    testDistributeShardsLikeWithHashSharding : function () {
+      let c1 = db._create(name1, { shardingStrategy: "hash", numberOfShards: 5 });
+      let c2 = db._create(name2, { distributeShardsLike: name1 });
+      assertEqual("hash", c1.properties()["shardingStrategy"]);
+      assertEqual("hash", c2.properties()["shardingStrategy"]);
+      assertEqual(5, c1.properties()["numberOfShards"]);
+      assertEqual(5, c2.properties()["numberOfShards"]);
+      assertEqual(["_key"], c1.properties()["shardKeys"]);
+      assertEqual(["_key"], c2.properties()["shardKeys"]);
+
+      for (let i = 0; i < 1000; ++i) {
+        c2.insert({ _key: "test" + i, value: i });
+      }
+
+      assertEqual([ 188, 192, 198, 204, 218 ], Object.values(c2.count(true)).sort());
+
+      for (let i = 0; i < 1000; ++i) {
+        assertEqual(i, c2.document("test" + i).value);
+      }
+    },
+
     testDistributeShardsLikeWithDiffentKeyAttribute : function () {
-      let c1 = db._create(name1, { shardingStrategy: "community-compat", numberOfShards: 5, shardKeys: ["one"] });
```

Comment:
Shouldn't all the existing tests for `community-compat` sharding also exist in the future?
I see no reason for changing them to use `hash` sharding now.
I think we should add **extra** tests for the hash sharding, but the existing tests for the other sharding types should remain there too.